idunn/geocoder/nlu_client.py

```diff
@@ -135,19 +177,36 @@ def fuzzy_match(cls, query, response):
             logger.error("Request to NLU tokenizer failed", exc_info=True)
             return []
 
-        intentions = []
         tags_list = [t for t in response_nlu.json()["NLU"] if t["tag"] != "O"]
-        tags_count = Counter(t["tag"] for t in tags_list)
-        if tags_count == {"city": 1, "cat": 1}:
-            # 1 category + 1 place
-            intention = await self.build_intention_category_place(tags_list, lang=lang)
-            if intention is not None:
-                intentions.append(intention)
-        elif tags_count == {"cat": 1}:
-            # 1 category
-            intention = await self.build_intention_category(tags_list, lang=lang)
-            if intention is not None:
-                intentions.append(intention)
+
+        if self.is_poi_request(tags_list):
+            return []
+
+        intentions = []
+        brand_query = self.build_brand_query(tags_list)
+        cat_query = self.build_category_query(tags_list)
+        place_query = self.build_place_query(tags_list)
+        skip_classifier = brand_query is not None
+
+        # If there is a label for both a category and a brand, we consider that the request is
+        # ambiguous and ignore both.
+        if bool(brand_query) ^ bool(cat_query):
+            if place_query:
+                # 1 category or brand + 1 place
+                # Brands are handled the same way categories except that we don't want to process
+                # them with the classifier.
+                intention = await self.build_intention_category_place(
+                    cat_query, place_query, lang=lang, skip_classifier=skip_classifier
+                )
+                if intention is not None:
+                    intentions.append(intention)
+            else:
+                # 1 category or brand
+                intention = await self.build_intention_category(
+                    cat_query, lang=lang, skip_classifier=skip_classifier
+                )
```

Comment:
`skip_classifier` is not a parameter for `build_intention_category`.  
It probably means one more test case should be added to cover that function.

Reply:
Oh, I missed this hack when witting tests: https://github.com/QwantResearch/idunn/blob/f5e00291a10c8bda992b7ea3f26069a2c98d2d39/tests/utils.py#L24.

I was afraid of adding more complicated code and it led me to forget about really testing brands :(

Is there a reason why we init PjSource globally? Maybe it could be more convenient to init it locally in modules where we use it? I suppose it would cost a few more http sessions which should be fine?

Reply:
We could probably initialize `PjSource` in different modules, but I am not sure it would make the tests any easier, given that settings would still be read on the first init only.